text/0000-location.md

```diff
@@ -0,0 +1,440 @@
+the transaction is invalid.
+
+The properties in the location will be swapped for the new properties and the 
+updated location will be set in state.
+
+The inputs for `LocationUpdateAction` must include:
+- Grid address of the Agent submitting the transaction
+- Grid address of the Organization the Location is being updated for
+- Grid address of the Location Namespace Schema the location’s properties must
+match
+- Grid address of the Location to be updated
+
+The outputs for `LocationUpdateAction` must include:
+- Grid address of the Location updated
+
+Note: This RFC handles updates to Location master data (e.g. a Location without 
+a defined inactivationDate is updated to have a defined inactivationDate). The 
+logic that consumes Location master data will decide what records are valid for 
+its use case. For example: A Location with a defined inactivationDate cannot be 
+used in new transactions but can still be used when referencing earlier 
+transactions. In this example, consuming logic will decide which Locations are 
+included in a transaction.
+
+### Location Delete Action
+
+`LocationDeleteAction` removes an existing location from state. The transaction 
+should be submitted by an agent, identified by its signing key, acting on behalf
+of the organization that corresponds to the owner in the location being updated.
+(Organizations and agents are defined by the Pike smart contract.)
+
+If the Grid setting grid.location.allow_delete is set to false, this transaction
+is invalid. The default value for grid.location.allow_delete is true. This 
+setting is stored using the Sawtooth Settings smart contract, more information
+can be found here.
+
+Validation requirements:
+- If a location with location_id does not exist the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must 
+belong to an organization in Pike state, otherwise the transaction is invalid.
+- The owner in the location must match the organization that the agent belongs 
+to, otherwise the transaction is invalid.
+- The agent must have the permission can_delete_location for the organization
+otherwise the transaction is invalid.
+
+The inputs for `LocationDeleteAction` must include:
+- Grid address of the Agent submitting the transaction
+- Grid address of the Organization the Location is being deleted for Grid 
+address of the Location to be deleted
+
+The outputs for `LocationDeleteAction` must include:
+- Grid address of the Location to be deleted
+
+### Defined GS1 Properties
+
+This Location RFC defines required and optional attributes (see below) for GS1
+physical locations such as manufacturing facilities, warehouses, distribution
+centers, retail stores, dock doors, etc. This implementation does not include
+attribution for specific industries, financial/tax account information, a 
+physical location extension, or digital locations and should be implemented 
```

Comment:
Why are digital locations being removed from a Grid system RFC ... where a big part of this is trying to bring supply chain to a standardized, digital place?

Reply:
Good question. In full transparency, excluding digital locations was a means to defer action on these until more discovery could be performed. Throughout my research in the standards I was unclear on whether digital GLNs carried different behaviors/attributes than other usage types (such as a physical location). GS1 clarified above that when a GLN is allocated there is not a required process to say "this is a physical location only", so your comment was a good nudge to broaden the scope. I updated the RFC to include all four usage types. I also added a comment in the RFC that additional attributes (e.g. for digital locations) could be integrated in the future as identified.